### PR TITLE
Change position of input on dragging

### LINKED ISSUE #7: change position of the text
user should be able to drag the text on the image to place it wherever they want

### PR DESCRIPTION
**Pull request by harshN-17**
fixed #7

### CODE REVIEW

File: src/components/Meme.js [hunk 1]
```diff
@@ -81,6 +88,56 @@ export default function Meme() {
...
+	//redering the text data in h2 tag 
+	const memeTexts = memeTextData.map(text => {
+		return <h2
+				key={text.id}
+				draggable
+				onDragStart={(e)=>handleDragStart(e, text.id)} 
+				style={text.style}
+				className="meme__text"
+			> 
+				{text.id === "1" ? meme.topText : meme.bottomText}
```
Comment (dev-lovedeep) [inline src/components/Meme.js:101]: map function also passes the index to the callback, so you can remove the id and instead use that index to work in place of the id.

another reason: we have one feature on adding multiple text, so in that case, we will not be considering this top and bottom text anymore

File: src/components/Meme.js [hunk 2]
```diff
@@ -22,6 +22,27 @@ export default function Meme() {
 		bottomText: "",
 		url: "",
 	});
+	
+	//data for all the input texts that are supposed to be draggable
+	const [memeTextData, setMemeTextData] = useState(
+		[
+			{
+				id: "1",
```
Comment (dev-lovedeep) [inline src/components/Meme.js:30]: remove id from here also

File: src/components/Meme.js [hunk 3]
```diff
@@ -81,6 +88,56 @@ export default function Meme() {
...
+		//get the xy coordinates of the mouse
+		let x = e.clientX;
+		let y = e.clientY;
+		//get the rectangle coordinates of the meme image.
+		const rect = document.querySelector('.meme').getBoundingClientRect();
+
+		//create a temp text array to make changes in the style and then set it to original state
+		const tempTexts = [...memeTextData]
+		//change the style of selected text and position it on the cursor
+		let index = tempTexts.findIndex((text) => text.id === id);
```
Comment (dev-lovedeep) [inline src/components/Meme.js:129]: you can remove this line also , once remove the use of id

File: src/components/Meme.js [hunk 4]
```diff
@@ -81,6 +88,56 @@ export default function Meme() {
...
+		if(index == -1) return;
+		tempTexts[index].style = {
+			//distance of text from left of image is equal to dist of the cursor from margin - meme image from left.
+			left: `${x-rect.left}px`,
+			//similarly calculate top.
+			top: `${y-rect.top}px`,
```
Comment (dev-lovedeep) [inline src/components/Meme.js:135]: it gives little inconsistent effect, check online to improve it, may try to consider the width of the text

**Comment by dev-lovedeep** [comment]
@harshN-17 you working on it?

File: src/components/Meme.js (hunk 4 above)
Comment (harshN-17) [reply src/components/Meme.js:135]: i cannot find a better way to implement the effect. can you help me out a little in it

Comment (dev-lovedeep) [reply src/components/Meme.js:135]: try this ,maybe this could help [link](https://www.w3schools.com/howto/howto_js_draggable.asp)

**Comment by harshN-17** [comment]
Check my latest commit

File: src/components/Meme.js [hunk 5]
```diff
@@ -117,22 +123,21 @@ export default function Meme() {
 		e.preventDefault();
 		//get the id that we set during dragstart
 		const id = e.dataTransfer.getData("id");
-		//get the xy coordinates of the mouse
-		let x = e.clientX;
-		let y = e.clientY;
-		//get the rectangle coordinates of the meme image.
-		const rect = document.querySelector('.meme').getBoundingClientRect();
+		//get the xy coordinates of the mouse when we drop the text
```
Comment (dev-lovedeep) [inline src/components/Meme.js:126]: this comment should be "relative change in the mouse position"

**Review by dev-lovedeep — COMMENTED** [review]
The drag is working well!! :tada: just fix the merge conflicts, and then I can merge your PR

**Comment by harshN-17** [comment]
I have fixed the merge conflicts as well as changed the comment as suggested. Please consider it for merge

**Comment by dev-lovedeep** [comment]
there is some issue present as of now also, but will be putting them as separate issue. good work :clinking_glasses:

**Pull request merged**